File: c++/Class/ClassClass.cpp


#include <iostream>

namespace sample {

class Test
{
	class SubTest
	{
	public:
		SubTest();

		int getInteger();
		int integer;
	};

public:
	int getInteger() { return this->subTest.getInteger(); }

private:
	SubTest subTest;
};

Test::SubTest::SubTest()
	: integer(3)
{
	
}

} // namespace sample

int sample::Test::SubTest::getInteger() {
	return this->integer;
}


int main()
{
	sample::Test test;

	std::cout << test.getInteger() << std::endl;

	return 0;
}

File: ruby/Class/ClassClass.rb


class Test
  class Sample
    attr :name
    
    def initialize(name)
      @name = name
    end
  end
  
  attr :sample
  def initialize()
    @sample = Sample.new("sample")
  end
end

test = Test.new

#p test.sample
#test.sample.name = "test"
p test.sample.name


